benchmarks/synchronization/synchronization.cpp

```diff
@@ -52,13 +52,13 @@ cuda_event_timer::cuda_event_timer(benchmark::State& state,
 }
 
 cuda_event_timer::~cuda_event_timer() {
-  RMM_CUDA_TRY(cudaEventRecord(stop, stream));
-  RMM_CUDA_TRY(cudaEventSynchronize(stop));
+  if (cudaEventRecord(stop, stream) != cudaSuccess) return;
```

Comment:
How about using asserts in the destructor so that we don't throw but we can still catch issues in a debug build?

Reply:
I wasn't sure if the `cudaEvent` calls were still needed to track timings in benchmark release-builds, but I'm happy to make them asserts instead.

Reply:
I think they are still needed, so you would just do the assert(cudaSuccess == result), not wrap the actual calls to EventRecord/Synchronize.